core/src/test/java/org/hibernate/ogm/backendtck/queries/QueriesWithEmbeddedCollectionTest.java

```diff
@@ -93,35 +96,40 @@ public void testEqualEmbeddedCollectionWithEmbeddableInCollectionWhereClause() t
 
 	@Test
 	public void testConjunctionOperatorWithEmbeddedInEmbeddedCollection() throws Exception {
-		List<?> result = session.createQuery( "FROM WithEmbedded e JOIN e.anEmbeddedCollection c WHERE c.item = 'item[1]' AND c.anotherItem IN ('secondItem[0]')" ).list();
+		List<?> result = session.createQuery( "FROM WithEmbedded e JOIN e.anEmbeddedCollection c WHERE c.item = 'item[0]' AND c.anotherItem IN ('secondItem[0]')" ).list();
 		assertThat( result ).onProperty( "id" ).containsOnly( 1L );
 	}
 
 	@Test
+	@SkipByGridDialect(value = GridDialectType.NEO4J)
```

Comment:
I think the error is the same as [OGM-801](https://hibernate.atlassian.net/browse/OGM-801), the nested embedded in a collection skip one level while the cypher query assumes that the mapping is correct. This causes an exception since they cannot found the element.

The error I fixed with this patch made this tests pass but now that it's fixed they don't work anymore. Not sure why it is working for the other datastores.